routers/web/user/setting/account.go

```diff
@@ -234,7 +234,11 @@ func DeleteEmail(ctx *context.Context) {
 func DeleteAccount(ctx *context.Context) {
 	ctx.Data["Title"] = ctx.Tr("settings")
 	ctx.Data["PageIsSettingsAccount"] = true
-
+	passwd := ctx.FormString("password")
+	if len(passwd) == 0 || !ctx.Doer.ValidatePassword(passwd) {
+		ctx.RenderWithErr(ctx.Tr("form.enterred_invalid_password"), tplSettingsAccount, nil)
+		return
+	}
```

Comment:
How does it work with external auth sources? eg: LDAP, the `passwd` field is empty.

(the first question is: should Gitea support deleting LDAP accounts?)

Reply:
@lunny Need your help on this.

Reply:
Maybe we need to check whether password is set before check popup the password dialog.

Reply:
You mean we need an api call from frontend that will check the password is set on the account?

Reply:
Thinking that,  there are still some edge cases remaining before we implement the solution around this 500 error.
I'm keeping this PR open for the further discussion and will raise a complete new PR.